From a4570559325a1a7abf1052b4c6b1d00fba2b7525 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 21 Mar 2018 12:01:39 +0000
Subject: [PATCH] Add a new UI event to let consumers know usysconf is running

This will help with the new Software Center where usysconf could potentially
be running for a while and it looks like the update has freezed, whereas in
reality the configuration is running.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 pisi/context.py | 6 ++++++
 pisi/ui.py      | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/pisi/context.py b/pisi/context.py
index 91c3348..8a556ee 100644
--- a/pisi/context.py
+++ b/pisi/context.py
@@ -81,6 +81,12 @@ def exec_usysconf():
         ui.error(_('usysconf not installed. Please upgrade!'))
         return
 
+    # Tell UI tools the system configuration is running
+    try:
+        ui.notify(pisi.ui.systemconf)
+    except:
+        pass
+
     try:
         os.system("{} run".format(usysconf_binary))
     except Exception as e:
diff --git a/pisi/ui.py b/pisi/ui.py
index 31a77ba..c472d49 100644
--- a/pisi/ui.py
+++ b/pisi/ui.py
@@ -13,7 +13,7 @@
 #
 
 (installed, upgraded, removed, installing, removing, configuring, configured, extracting,
- downloading, packagestogo, updatingrepo, cached, desktopfile)  = range(13)
+ downloading, packagestogo, updatingrepo, cached, desktopfile, systemconf)  = range(14)
 
 class UI(object):
     "Abstract class for UI operations, derive from this."
-- 
2.16.2

